benchmarks/helics/WattsStrogatzFederate.hpp

```diff
@@ -0,0 +1,206 @@
+            exit(1);
+        }
+
+        if (app->get_option("--set_rand_seed")->count() == 0) {
+            std::mt19937 random_engine(0x600d5eed);
+            std::uniform_int_distribution<unsigned int> rand_seed_uniform;
+            for (int i = 0; i < index; i++) {
+                rand_seed_uniform(random_engine);
+            }
+            setRandomSeed(rand_seed_uniform(random_engine));
+        }
+
+        // set up based on given params
+        // en.cppreference.com/w/cpp/numeric/random/exponential_distribution
+        if (generateRandomSeed) {
+            std::random_device rd;
+            rand_gen.seed(rd());
+        } else {
+            rand_gen.seed(seed);
+        }
+        rand_rewire = std::uniform_real_distribution<double>(0.0, 1.0);
+        // create random number distribution for rewiring links (there are maxIndex-k-2 available
+        // links)
+        if (maxIndex - k > 1) {
+            rand_available_link = std::uniform_int_distribution<unsigned int>(0, maxIndex - k - 2);
+        }
+        // create random number distribution for picking a link to transmit on
+        rand_transmit_link = std::uniform_int_distribution<unsigned int>(0, k - 1);
+    }
+
+    void doFedInit() override
+    {
+        // Construct the Watts-Strogatz graph.
+        // For benchmark setup reasons from within the Federates, it is a directed graph (typically
+        // it is undirected). The doParamInit check on k < maxIndex - 1 ensures that currentEdges
+        // and availableEdges will not include our own index, avoiding self-loops.
+        std::vector<unsigned int> currentEdges;
+        std::vector<unsigned int> availableEdges;
+        currentEdges.reserve(k);
+        availableEdges.reserve(maxIndex - k - 1);
+
+        // Construct half of a ring lattice
+        for (int i = 1; i <= k; i++) {
+            currentEdges.push_back((index + i) % maxIndex);
+        }
+
+        // Create the set of links available as rewiring choices
+        for (int i = 1; i <= maxIndex - k - 1; i++) {
+            auto edge = index - i;
+            if (edge < 0) {
+                edge += maxIndex;
+            }
+            availableEdges.push_back(edge);
+        }
+
+        // Re-wire links
+        if (!availableEdges.empty()) {
+            // NOLINTNEXTLINE(modernize-loop-convert)
+            for (unsigned int i = 0; i < currentEdges.size(); i++) {
+                // Decide if the link should be rewired or not
+                if (rand_rewire(rand_gen) < b) {
+                    // Criteria for the new link:
+                    // 1. Avoid self-loops
+                    // 2. Avoid duplication with links that exist at this point in the algorithm
+
+                    // Pick a new link from the available edges
+                    auto newLink = rand_available_link(rand_gen);
+
+                    // Swap current link with available edge
+                    auto tmp = availableEdges[newLink];
```

Comment:
can you use std::swap here?

Reply:
It looks like std::swap isn't changing the data in the arrays.